webrtc.html

```diff
@@ -9671,6 +9691,59 @@ <h2>Dictionary <a class="idlType">RTCStats</a> Members</h2>
       from RTCStats that they indicate, are documented in
       [[!WEBRTC-STATS]].</p>
     </section>
+    <section>
+      <h3><dfn>RTCStatsLifetimeEvent</dfn></h3>
+      <p>The <code><a>statslifetimeended</a></code> event uses
+        the <code><a>RTCStatsLifetimeEvent</a></code>.</p>
+      <div>
+        <pre class="idl">
+          [ Constructor (DOMString type, RTCStatsLifetimeEventInit
+          eventInitDict), Exposed=Window]
+          interface RTCStatsLifetimeEvent : Event {
+            readonly attribute any stat; // Must be RTCStats subtype
```

Comment:
Should we change this to sequence&lt;any&gt;?

For example if there is both a sender stat and RTP stream stat that are simultaneously removed by stopping a transceiver (maybe a bad example, if transceivers are never removed then I guess neither are these objects). Or any other example where two objects referencing each other are both removed due to the same event.

I'm thinking it is a problem if a stats object references an object that neither exists amongst "deleted stats" (because that event has not fired yet) or "current stats" (because it has already been removed and event queued). I will refer to a stats object that has been deleted but its event has not been processed yet as a "limbo stats object", stuck between being added to the application's "deleted stats" list and the PC's "current stats".

On the other hand it might not be a problem. Because if you want to do lookup in both "deleted stats" and "current stats" you have to call getStats() to get the "current stats", and by the time you have the "current stats" the limbo stats object's event will already have fired and the object added to "deleted stats".

```
deletedStats = new Set();
pc.onstatslifetimeended = e => {
  deletedStats.add(e.stat);
  // Trying to process e.stat by looking up references to other stats objects in deletedStats
  // would be a problem for "limbo stats".

  pc.getStats().then(r => {
    currentStats = r;
    // Process e.stat, it may reference things in both deletedStats and currentStats.
    // By the time we reach this line, deletedStats will contain any "limbo stats".
  });
}
```

Maybe we don't need to change it to a sequence? On the other hand you might want to process it without forcing a getStats() call. You can of course always do setTimeout() and you're safe, but it's not obvious to developers and it might be considered a hack?

This should be discussed before merge. If we decide to not do a sequence we need to write a comment about this in the description.

Reply:
I don't like having to defensively getStats() because "what if it references an object still in there", that means if 10 objects get deleted we may do 10 getStats(). setTimeout() is also hacky, because there is no way to tell "was this the last event?" you would do things multiple times here as well.

You could try to deduce if multiple events belong together by looking at their timestamp, but multiple objects deleted at the same time might not have the same timestamp. Some objects have timestamp for the last SR or RR received, not for when the object was deleted.

I think we need a `sequence<any>` to inform the application of all things removed associated with the same removal event.

Reply:
`any` is too broad. We should at least use `object` since it is guaranteed to be an object.

I agree with @henbos we should return all simultaneously deleted stats at once rather than fire multiple events. But why add another way to enumerate stats?

Why don't we simply return an [RTCStatsReport](http://w3c.github.io/webrtc-pc/#dom-rtcstatsreport)? That'd be consistent and it'd work as an attribute.